src/api.jl

```diff
@@ -0,0 +1,24 @@
+module api
+
+paths = ByteString[]
+const libvulkan = Libdl.find_library(["libvulkan", "vulkan", "libvulkan.so.1"], paths)
+@assert libvulkan != ""
+
+function _vk_func(func, ret_type, arg_types)
+  local args_in = Symbol[symbol("arg$i::$T")
+  for (i, T) in enumerate(arg_types.args)]
+  esc(quote
+    function $func($(args_in...))
+      ccall(($(string(func)), libvulkan),
+      $ret_type,
+      $arg_types,
+      $(args_in...))
+    end
+  end)
+end
```

Comment:
can we substitues `ptr` with `Ref` in the ccall definition? So that we have:

```julia
function test(arr :: Ref{Int64})
  ccall(:test, lib, Void, (Ref{Int64},), arr)
end
```